src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -15,22 +27,30 @@
     public static final String COMMAND_WORD = "find";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all patients whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+            + "the specified keywords (case-insensitive),"
+            + "or whose phone number or nric matches and displays them as a list with index numbers.\n"
+            + "Parameters: [" + PREFIX_NAME + "NAME [MORE_NAMES]] "
+            + "[" + PREFIX_NRIC + "NRIC [MORE_NRICS]] "
+            + "[" + PREFIX_PHONE + "PHONE_NUMBER [MORE_PHONE_NUMBERS]]\n"
+            + "Example: " + COMMAND_WORD + " n/alex p/99998888 i/S1234567I";
 
-    private final NameContainsKeywordsPredicate predicate;
+    public static final String MESSAGE_NO_FIND = "At least one field to edit must be provided";
+    private final FindPatientDescriptor descriptor;
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    public FindCommand(FindPatientDescriptor descriptor) {
+        this.descriptor = descriptor;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
+        assert descriptor.getNamePredicate().isPresent();
+        assert descriptor.getNricPredicate().isPresent();
+        assert descriptor.getPhonePredicate().isPresent();
```

Comment:
I'm not sure about this but wouldn't the assertion fail if certain prefix is not used?  
E.g. if name prefix is unused, wouldn't there be no namePredicate?

Reply:
Thanks for bring this up!